ui/app/src/views/projects/ProjectTabs.tsx

```diff
@@ -0,0 +1,210 @@
+  const [openCreateVariableDrawerState, setOpenCreateVariableDrawerState] = useState(false);
+
+  const handleDashboardCreation = (dashboardSelector: DashboardSelector) => {
+    navigate(`/projects/${dashboardSelector.project}/dashboards/${dashboardSelector.dashboard}/create`);
+  };
+
+  const handleVariableCreation = useCallback(
+    (variable: VariableResource) => {
+      createVariableMutation.mutate(variable, {
+        onSuccess: (updatedVariable: VariableResource) => {
+          successSnackbar(`Variable ${getVariableExtendedDisplayName(updatedVariable)} have been successfully created`);
+          setOpenCreateVariableDrawerState(false);
+        },
+        onError: (err) => {
+          exceptionSnackbar(err);
+          throw err;
+        },
+      });
+    },
+    [exceptionSnackbar, successSnackbar, createVariableMutation]
+  );
+
+  switch (props.index) {
+    case dashboardTabIndex:
+      return (
+        <>
+          <CRUDButton
+            text="Add Dashboard"
+            variant="contained"
+            onClick={() => setOpenCreateDashboardDialogState(true)}
+          />
+          <CreateDashboardDialog
+            open={openCreateDashboardDialogState}
+            projectOptions={[props.projectName]}
+            onClose={() => setOpenCreateDashboardDialogState(false)}
+            onSuccess={handleDashboardCreation}
+          />
+        </>
+      );
+    case variablesTabIndex:
+      return (
+        <>
+          <CRUDButton text="Add Variable" variant="contained" onClick={() => setOpenCreateVariableDrawerState(true)} />
+          <VariableFormDrawer
+            variable={{
+              kind: 'Variable',
+              metadata: {
+                name: 'NewVariable',
+                project: props.projectName,
+              },
+              spec: {
+                kind: 'TextVariable',
+                spec: {
+                  name: 'NewVariable',
+                  value: '',
+                },
+              },
+            }}
+            isOpen={openCreateVariableDrawerState}
+            onChange={handleVariableCreation}
+            onClose={() => setOpenCreateVariableDrawerState(false)}
+            action="create"
+          />
+        </>
+      );
+    case datasourcesTabIndex:
+      return (
+        <CRUDButton text="Add Datasource" variant="contained" onClick={() => setOpenCreateDashboardDialogState(true)} />
+      );
+    default:
+      return <></>;
+  }
+}
+
+interface TabPanelProps {
+  children?: ReactNode;
+  index: string;
+  value: string;
+}
+
+function TabPanel(props: TabPanelProps) {
+  const { children, value, index, ...other } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`project-tabpanel-${index}`}
+      aria-labelledby={`project-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Box sx={{ paddingTop: 2 }}>
+          <Typography>{children}</Typography>
```

Comment:
Not sure why you passed the children to a Typography component here? But the interpreter doesn't like it (_"Warning: validateDOMNesting(...): \<div\> cannot appear as a descendant of \<p\>"_ in console)

Reply:
Oops... 😅 Fixed